### PR TITLE
Upgrade e2e test dependencies and Mocha

### PR DESCRIPTION
**Pull request by mattseddon**
🎄 🎅🏻 💩 
----

This PR gets us past the breaking change in `wdio-vscode-service`. In order to do this the end to end test suite has to become and ESM module (sub-package). This means that it cannot call any of the rest of the codebase directly (which is probably a good thing).

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #!/bin/sh
 . "$(dirname $0)/_/husky.sh"
 
-yarn run tsc -p tsconfig.json --noEmit
```
Comment (mattseddon) [inline .husky/pre-commit:4]: [F] This now throws some wacky errors

Comment (mattseddon) [reply .husky/pre-commit:4]: Can safely replace with the command on the right or drop altogether. Makes me think we no longer need the `tsconfig.json` at the root of the monorepo.

File: extension/src/test/e2e/extension.test.ts [hunk 2]
```diff
@@ -1,5 +1,4 @@
 import { join } from 'path'
```
Comment (mattseddon) [inline extension/src/test/e2e/extension.test.ts:1]: [F] Most of the changes are indentation based (dropped the suite statement as it no longer worked). Will mark the actual changes and give reasons.

File: extension/src/test/e2e/util.ts [hunk 3]
```diff
@@ -1,10 +1,10 @@
 import {
   ChainablePromiseArray,
   ChainablePromiseElement,
-  ElementArray
+  ElementArray,
+  Key
 } from 'webdriverio'
 import { ViewControl } from 'wdio-vscode-service'
-import { delay } from '../../util/time'
```
Comment (mattseddon) [inline extension/src/test/e2e/util.ts:7]: [F] Good thing the new v8 API has it's own delay as we can't import anything from our own repo here (not ESM).

File: extension/src/test/e2e/extension.test.ts [hunk 4]
```diff
@@ -11,184 +10,190 @@ import {
   runModifiedExperiment,
   waitForDvcToFinish,
   waitForViewContainerToLoad
-} from './util'
-import { ExperimentsWebview } from './pageObjects/experimentsWebview'
-import { PlotsWebview } from './pageObjects/plotsWebview'
-
-suite('DVC Extension For Visual Studio Code', () => {
```
Comment (mattseddon) [inline extension/src/test/e2e/extension.test.ts:18]: [F] Can no longer use suite.

File: extension/src/test/e2e/extension.test.ts [hunk 5]
```diff
@@ -11,184 +10,190 @@ import {
...
-
-      expect(finalRows.length).toStrictEqual(initialRows + epochs)
-      await webview.unfocus()
-      await waitForDvcToFinish()
-      const workbench = await browser.getWorkbench()
-      await workbench.executeCommand('Terminal: Kill All Terminals')
-    }).timeout(180000)
+    expect(await webview.table$$).toHaveLength(1)
+
+    return webview.unfocus()
```
Comment (mattseddon) [inline extension/src/test/e2e/extension.test.ts:63]: [F] This test has to be split in half or the framework cannot find the command palette.

File: extension/src/test/e2e/extension.test.ts [hunk 6]
```diff
@@ -11,184 +10,190 @@ import {
...
   })
 
-  describe('Plots Webview', () => {
-    before(async () => {
-      await closeAllEditors()
-    })
+  it('should show the correct number of records before running an experiment', async function () {
```
Comment (mattseddon) [inline extension/src/test/e2e/extension.test.ts:66]: [F] This test has to be split in half or the framework cannot find the command palette.

**Review by julieg18 — APPROVED** [review]
> 🎄 🎅🏻 💩

Ah yes... Christmas tree santa poop..

Code looks good though! Great work!

**Pull request merged**